cocos/physics/framework/components/character-controllers/character-controller.ts

```diff
@@ -0,0 +1,521 @@
+    }
+
+    public set slopeLimit (value) {
+        if (this._slopeLimit === value) return;
+        this._slopeLimit = Math.abs(value);
+        if (this._cct) {
+            this._cct.setSlopeLimit(value);
+        }
+    }
+
+    /**
+     * @en
+     * Gets or sets the contact offset of the character controller.
+     * @zh
+     * 获取或设置角色控制器的接触间隙。
+     */
+    @tooltip('i18n:physics3d.character_controller.contactOffset')
+    @type(CCFloat)
+    public get contactOffset () {
+        return this._contactOffset;
+    }
+
+    public set contactOffset (value) {
+        if (this._contactOffset === value) return;
+        this._contactOffset = Math.abs(value);
+        if (this._cct) {
+            this._cct.setContactOffset(value);
+        }
+    }
+
+    // /**
+    //  * @en
+    //  * Gets or sets if the character controller can collide with other objects.
+    //  * @zh
+    //  * 获取或设置角色控制器是否和发生碰撞。
+    //  */
+    // @tooltip('i18n:physics3d.character_controller.detectCollisions')
+    // @type(CCBoolean)
+    // public get detectCollisions () {
+    //     return this._detectCollisions;
+    // }
+
+    // public set detectCollisions (value) {
+    //     if (this._detectCollisions === value) return;
+    //     this._detectCollisions = value;
+    //     if (this._cct) {
+    //         this._cct.setDetectCollisions(value);
+    //     }
+    // }
+
+    // /**
+    //  * @en
+    //  * Gets or sets if the character controller enables overlap recovery when penetrating with other colliders.
+    //  * @zh
+    //  * 获取或设置角色控制器和其他碰撞体穿透时是否恢复。
+    //  */
+    // @tooltip('i18n:physics3d.character_controller.enableOverlapRecovery')
+    // @type(CCBoolean)
+    // public get enableOverlapRecovery () {
+    //     return this._enableOverlapRecovery;
+    // }
+
+    // public set enableOverlapRecovery (value) {
+    //     if (this._enableOverlapRecovery === value) return;
+    //     this._enableOverlapRecovery = value;
+    //     if (this._cct) {
+    //         this._cct.setOverlapRecovery(value);
+    //     }
+    // }
+
+    /**
+     * @en
+     * Gets or sets the center of the character controller in local space.
+     * @zh
+     * 获取或设置角色控制器的中心点在局部坐标系中的位置。
+     */
+    @tooltip('i18n:physics3d.character_controller.center')
+    @type(Vec3)
+    public get center () {
+        return this._center;
+    }
+
+    public set center (value: Vec3) {
+        if (this._center === value) return;
+        Vec3.copy(this._center, value);
+        if (this._cct) { //update cct position
+            Vec3.copy(VEC3_0, this.node.worldPosition);
+            Vec3.multiply(scaledCenter, this.center, this.node.worldScale);
+            VEC3_0.add(scaledCenter);//cct world position
+            this._cct.setPosition(VEC3_0);
+        }
+    }
+
+    /**
+     * @en
+     * Gets the wrapper object, through which the lowLevel instance can be accessed.
+     * @zh
+     * 获取封装对象，通过此对象可以访问到底层实例。
+     */
+    public get characterController () {
```

Comment:
We should not expose the underlying objects, otherwise any future upgrades will lead to compatibility issues.

Reply:
ok